src/storage/UpdateEdgeProcessor.cpp

```diff
@@ -0,0 +1,373 @@
+    auto ret = kvstore_->prefix(spaceId_, partId, prefix, &iter);
+    if (ret != kvstore::ResultCode::SUCCEEDED) {
+        VLOG(3) << "Error! ret = " << static_cast<int32_t>(ret) << ", spaceId " << spaceId_;
+        return ret;
+    }
+    // Only use the latest version.
+    if (iter && iter->valid()) {
+        key_ = iter->key().toString();
+        edgeReader_ = RowReader::getEdgePropReader(schemaMan_,
+                                                   iter->val(),
+                                                   spaceId_,
+                                                   edgeKey.edge_type);
+        filterReader_ = RowReader::getEdgePropReader(schemaMan_,
+                                                     iter->val(),
+                                                     spaceId_,
+                                                     edgeKey.edge_type);
+    } else if (insertable_) {
+        resp_.set_upsert(true);
+        auto now = std::numeric_limits<int64_t>::max() - time::WallClock::fastNowInMicroSec();
+        key_ = NebulaKeyUtils::edgeKey(partId, edgeKey.src, edgeKey.edge_type,
+                                       edgeKey.ranking, edgeKey.dst, now);
+        auto schema = this->schemaMan_->getEdgeSchema(spaceId_, edgeKey.edge_type);
+        RowWriter writer(schema);
+        edgeReader_ = RowReader::getRowReader(writer.encode(), schema);
+        filterReader_ = RowReader::getRowReader(writer.encode(), schema);
+    } else {
+        VLOG(3) << "Missed partId " << partId << ", edge " << edgeKey.src << " --> " << edgeKey.dst;
+        return kvstore::ResultCode::ERR_KEY_NOT_FOUND;
+    }
+    this->collectProps(edgeReader_.get(), key_, props, fcontext, collector);
+    return ret;
+}
+
+
+kvstore::ResultCode UpdateEdgeProcessor::processEdge(PartitionID partId,
+                                                     const cpp2::EdgeKey& edgeKey) {
+    FilterContext fcontext;
+    RowWriter writer(nullptr);
+    PropsCollector collector(&writer);
+    auto ret = collectEdgesProps(partId,
+                                 edgeKey,
+                                 this->edgeContext_.props_,
+                                 &fcontext,
+                                 &collector);
+    if (ret != kvstore::ResultCode::SUCCEEDED) {
+        return ret;
+    }
+    for (auto& tc : this->tagContexts_) {
+        VLOG(3) << "partId " << partId << ", vId " << edgeKey.src
+                << ", tagId " << tc.tagId_ << ", prop size " << tc.props_.size();
+        ret = this->collectVertexProps(partId,
+                                       edgeKey.src,
+                                       tc.tagId_,
+                                       tc.props_,
+                                       &fcontext,
+                                       &collector);
+        if (ret != kvstore::ResultCode::SUCCEEDED) {
+            return ret;
+        }
+    }
+
+    // the Where clause for filting
```

Comment:
The filter is not used for where clause?

Reply:
Here is  checkFilter(), rather than traversing filter.

Reply:
It has put into the storage layer.